### PR TITLE
Se 571 remove legacy token example contracts and create deployment doc

### PR DESCRIPTION
**Pull request by mpetersoCode55**
Removed Legacy Token Examples
Added IProtocolTokenMin interface
Added Protocol Token Minimal Implementation document
Added integration tests for base implementations (ERC20 and ERC721) of the min interface

### CODE REVIEW

**Review by TJ-Everett — APPROVED** [review]
This was amazing work Michael. Great job!

File: docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+function connectHandlerToToken(address _handlerAddress) external appAdministratorOnly(appManagerAddress)
+```
+
+- getHandlerAddress: used to retrieve the address of the Protocol Token Handler.
+
+```c
+function getHandlerAddress() external view override returns (address)
+```
+
+## CheckAllRules Hook
```
Comment (ShaneDuncan602) [inline docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md]: Probably want the case to match, checkAllRules

File: docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+The token must also override the _beforeTokenTransfer function and add a call to the CheckAllRules hook defined in the IProtocolTokenHandler interface (using the handler address set in the above functions).
```
Comment (ShaneDuncan602) [inline docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md]: same

File: src/client/token/IProtocolTokenMin.sol
Comment (ShaneDuncan602) [inline src/client/token/IProtocolTokenMin.sol:1]: Nice and clean. I love it

**Review by ShaneDuncan602 — APPROVED** [review]
Fantastic. One small suggestion on using mixed case in the markdown

File: docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md (hunk 1 above)
Comment (mpetersoCode55) [reply docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md]: Resolved

File: docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md (hunk 2 above)
Comment (mpetersoCode55) [reply docs/userGuides/Architecture/Client/PROTOCOL_TOKEN_MINIMAL_IMPLEMENTATION.md]: Resolved

**Review by ShaneDuncan602 — APPROVED** [review]

**Pull request merged**